python/cudf/cudf/core/column/column.py

```diff
@@ -2533,6 +2533,20 @@ def _construct_array(
         if inferred_dtype == "interval":
             # Only way to construct an Interval column.
             return pd.array(arbitrary)
+        elif (
+            inferred_dtype == "string" and getattr(dtype, "kind", None) == "M"
+        ):
+            # We may have date-like strings with timezones
+            try:
+                pd_arbitrary = pd.to_datetime(arbitrary)
+            except pd.errors.OutOfBoundsDatetime:
+                # Not an issue in pandas>=2.0
```

Comment:
Does this comment mean that this error won't be raised when we upgrade to pandas 2.0? If so, one suggestion I would have is that we implement a utility function that will raise an exception everywhere that we have a pandas<2.0 code path if the detected pandas version is 2.0. Then when we actually upgrade pandas we can validate that we changed everywhere that needed changing.

Reply:
Ah so actually this doesn't work yet in pandas (it was related to non-nanosecond support in pandas 2.0, but it's not implemented everywhere yet). I linked the issue to this exception path